Documentation/teaching/lectures/interrupts.rst

```diff
@@ -494,22 +591,60 @@ Deferrable actions
       * runs in interrupt context
       * can be dinamically allocated
       * same handler runs are serialized
+
     * workqueues
-   
+
       * run in process context
 
-.. slide:: Soft IRQ
+Deferrable actions have APIs to: **initialize** an instance, **activate** or
+**schedule** the action and **mask/disable** and **unmask/enable** the execution
+of the callback function. The later is used for synchronization purposes between
+the callback function and other contexts.
+
+Soft IRQs
+---------
+
+Soft IRQs is the term used for the low level mechanism that implements deferring
+work from interrupt handlers but that still runs in interrupt context.
+
+.. slide:: Soft IRQs
    :inline-contents: True
    :level: 2
 
-    * init: open_softirq()
-    * activation: raise_softirq()
-    * execution: do_softirq()
-    * it runs
+    Soft IRQ APIs:
+
+      * initialize: :c:func:`open_softirq`
+      * activation: :c:func:`raise_softirq`
+      * masking: :c:func:`local_bh_disable`, :c:func:`local_bh_enable`
 
-      * after an interrupt handler
+    Once activated, the callback function :c:func:`do_softirq` runs either:
+
+      * after an interrupt handler or
       * from the ksoftirqd kernel thread
 
+
+.. slide:: ksoftirqd
+   :inline-contents: False
+   :level: 2
+
+    * minimum priority kernel thread
+    * runs softirqs after certain limits are reached
+    * tries to achieve good latency and avoid process starvation
+
+
+Since softirqs can reschedule themselves or other interrupts can occur that
+reschedules them, they can potentially lead to (temporary) process starvation if
+checks are not put into place. Currently, the Linux kernel does not allow
+running soft irqs for more than :c:macro:`MAX_SOFTIRQ_TIME` or rescheduling for
+more than :c:macro:`MAX_SOFTIRQ_RESTART` consecutive times.
+
+Once these limits are reached a special kernel thread, **ksoftirqd** is wake-up
+and all of the rest of pending soft irqs will be run from the context of this
```

Comment:
... all pending softirqs ... will be re-enabled by **ksoftirqd** (I wouldn't use "run" or "context")